kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionUtils.java

```diff
@@ -87,46 +90,78 @@ public boolean getBoolean(int rowId) {
         } else if (dataType instanceof ByteType) {
-            compareByte(left, right, result);
+            valueComparator = new ByteValueComparator(left, right, vectorComparator);
         } else if (dataType instanceof ShortType) {
-            compareShort(left, right, result);
+            valueComparator = new ShortValueComparator(left, right, vectorComparator);
         } else if (dataType instanceof IntegerType || dataType instanceof DateType) {
-            compareInt(left, right, result);
+            valueComparator = new IntegerValueComparator(left, right, vectorComparator);
         } else if (dataType instanceof LongType ||
                 dataType instanceof TimestampType ||
                 dataType instanceof TimestampNTZType) {
-            compareLong(left, right, result);
+            valueComparator = new LongValueComparator(left, right, vectorComparator);
         } else if (dataType instanceof FloatType) {
-            compareFloat(left, right, result);
+            valueComparator = new FloatValueComparator(left, right, vectorComparator);
         } else if (dataType instanceof DoubleType) {
-            compareDouble(left, right, result);
+            valueComparator = new DoubleValueComparator(left, right, vectorComparator);
         } else if (dataType instanceof DecimalType) {
-            compareDecimal(left, right, result);
+            valueComparator = new DecimalValueComparator(left, right, vectorComparator);
         } else if (dataType instanceof StringType) {
-            compareString(left, right, result);
+            valueComparator = new StringValueComparator(left, right, vectorComparator);
         } else if (dataType instanceof BinaryType) {
-            compareBinary(left, right, result);
+            valueComparator = new BinaryValueComparator(left, right, vectorComparator);
         } else {
             throw new UnsupportedOperationException(dataType + " can not be compared.");
         }
-        return result;
+
+        return new ColumnVector() {
+
+            @Override
+            public DataType getDataType() {
+                return BooleanType.BOOLEAN;
+            }
+
+            @Override
+            public void close() {
+                Utils.closeCloseables(left, right);
+            }
+
+            @Override
+            public int getSize() {
+                return left.getSize();
+            }
+
+            @Override
+            public boolean isNullAt(int rowId) {
+                return left.isNullAt(rowId) || right.isNullAt(rowId);
+            }
+
+            @Override
+            public boolean getBoolean(int rowId) {
+                if (isNullAt(rowId)) {
+                    return false;
+                }
+                return valueComparator.getCompareResult(rowId);
+            }
+        };
     }
 
     static void compareBoolean(ColumnVector left, ColumnVector right, int[] result) {
```

Comment:
we can remove all these methods right as they are no longer used?

Reply:
fixed.